web/src/settings_components.ts

```diff
@@ -426,6 +443,26 @@ export function read_field_data_from_form(
     return undefined;
 }
 
+function get_field_data_input_value($input_elem: JQuery): string | undefined {
+    const $profile_field_form = $input_elem.parent();
+    const profile_field_id = Number.parseInt(
+        $($profile_field_form).attr("data-profile-field-id")!,
+        10,
+    );
+    const field = realm.custom_profile_fields.find((field) => field.id === profile_field_id)!;
+
+    if (field.field_data === "") {
+        return "";
```

Comment:
It was necessary because the parent `span` element for field_data was always visible. I put the `span` to only be present if either of `is_select_field` or `is_external_account_field` is true. Now it won't be necessary. Changes at [diff link](https://github.com/zulip/zulip/compare/b0d08f767aad67ff0c691530e0b53881256a6c42..fd0f742681283a3ec7af684e59b34ce788c720fe).